File: calc/javacalc/src/main/java/com/javacalc/Operation.java
package com.javacalc;

public enum Operation {
    ADD("+", "＋"),
    SUBTRACT("-", "－"),
    MULTIPLY("*", "×"),
    DIVIDE("/", "÷");

    private final String symbol;
    private final String displaySymbol;

    Operation(String symbol, String displaySymbol) {
        this.symbol = symbol;
        this.displaySymbol = displaySymbol;
    }

    public String getSymbol() {
        return symbol;
    }

    public String getDisplaySymbol() {
        return displaySymbol;
    }

    public double apply(double a, double b) {
        return switch (this) {
            case ADD -> a + b;
            case SUBTRACT -> a - b;
            case MULTIPLY -> a * b;
            case DIVIDE -> {
                if (b == 0) {
                    throw new ArithmeticException("Division by zero");
                }
                yield a / b;
            }
        };
    }

    public static Operation fromSymbol(String symbol) {
        for (Operation op : values()) {
            if (op.symbol.equals(symbol)) {
                return op;
            }
        }
        return null;
    }
}

File: calc/fluttercalc/lib/operation.dart
enum Operation {
  add('+', '＋'),
  subtract('-', '－'),
  multiply('*', '×'),
  divide('/', '÷');

  const Operation(this.symbol, this.displaySymbol);

  final String symbol;
  final String displaySymbol;

  double apply(double a, double b) {
    switch (this) {
      case Operation.add:
        return a + b;
      case Operation.subtract:
        return a - b;
      case Operation.multiply:
        return a * b;
      case Operation.divide:
        if (b == 0) {
          throw ArgumentError('Division by zero');
        }
        return a / b;
    }
  }

  static Operation? fromSymbol(String symbol) {
    for (final op in Operation.values) {
      if (op.symbol == symbol) return op;
    }
    return null;
  }
}
